cms/djangoapps/contentstore/views/course.py

```diff
@@ -1058,6 +1059,19 @@ def settings_handler(request, course_key_string):
             sidebar_html_enabled = course_experience_waffle().is_enabled(ENABLE_COURSE_ABOUT_SIDEBAR_HTML)
             # self_paced_enabled = SelfPacedConfiguration.current().enabled
 
+            course_category_nodes = CourseCategory.objects.filter(parent=None)
+            course_category_options = []
+
+            def add_nodes(nodes, prefix):
```

Comment:
I myself do not like it when a function is declared in a function if it is not decorators. It is dirty and usually done quickly.
In a specific case, it is a recursive function that traverses a tree, takes a list from an external context, and changes it without returning anything. I don’t really like that. But I put it in a separate module. I did not rewrite since this is a recursion for traversing a tree and I did not want to spend the time for debugging, but I would have to debug there 🙂